src/info/manifest.c

```diff
@@ -383,6 +386,750 @@ manifestNewInternal(void)
     FUNCTION_TEST_RETURN(this);
 }
 
+/**********************************************************************************************************************************/
+typedef struct ManifestBuildData
+{
+    Manifest *manifest;
+    const Storage *storagePg;
+    const String *tablespaceId;                                     // Tablespace id if PostgreSQL version has one
+    bool online;                                                    // Is this an online backup?
+    const String *manifestWalName;                                  // Wal manifest name for this version of PostgreSQL
+    RegExp *dbPathExp;                                              // Identify paths containing relations
+    RegExp *tempRelationExp;                                        // Identify temp relations
+    RegExp *standbyExp;                                             // Identify files that must be copied from the primary
+    StringList *excludeContent;                                     // Exclude contents of directories
+    StringList *excludeSingle;                                      // Exclude a single file/link/path
+
+    // These change with each level of recursion
+    const String *manifestParentName;                               // Manifest name of this file/link/path's parent
+    const String *pgPath;                                           // Current path in the PostgreSQL data directory
+    bool dbPath;                                                    // Does this path contain relations?
+} ManifestBuildData;
+
+// Callback to process files/links/paths and add them to the manifest
+static void
+manifestBuildCallback(void *data, const StorageInfo *info)
+{
+    FUNCTION_TEST_BEGIN();
+        FUNCTION_TEST_PARAM_P(VOID, data);
+        FUNCTION_TEST_PARAM(STORAGE_INFO, *storageInfo);
+    FUNCTION_TEST_END();
+
+    ASSERT(data != NULL);
+    ASSERT(info != NULL);
+
+    // Skip all . paths because they have already been recorded on the previous level of recursion
+    if (strEq(info->name, DOT_STR))
+    {
+        FUNCTION_TEST_RETURN_VOID();
+        return;
+    }
+
+    // Skip any path/file/link that begins with pgsql_tmp.  The files are removed when the server is restarted and the directories
+    // are recreated.
+    if (strBeginsWithZ(info->name, PG_PREFIX_PGSQLTMP))
+    {
+        FUNCTION_TEST_RETURN_VOID();
+        return;
+    }
+
+    // Get build data
+    ManifestBuildData buildData = *(ManifestBuildData *)data;
+    unsigned int pgVersion = buildData.manifest->data.pgVersion;
+
+    // Contruct the name used to identify this file/link/path in the manifest
+    strPtr(info->name);
+    strPtr(buildData.manifestParentName);
```

Comment:
Probably I had them there to avoid compiler errors while writing code, kind of like `(void)info`.  Probably.

But they are clearly vestigial now and should be removed.

Reply:
Removed (commit 1fe22c3bb51dbb7070d22d8329ee999290ea8560)